plugins/sharding/logical_range_filter.rb

```diff
@@ -1017,70 +1018,73 @@ def filter_by_range(range_index, min, min_border, max, max_border)
           lexicon = range_index.domain
           @context.referred_objects << lexicon
           data_table = range_index.range
```

Comment:
`data_table` is same to `shard.table` in some cases.
In those cases, if `data_table` is added to `@context.referred_objects`, some `shard.table`s are not un-referenced until the end of command.

So I have changed it to be un-referenced immdiately but this may make performance worse.
I'm thinking about a good solution...